### PR TITLE
New confusion natrix

### PR DESCRIPTION
**Pull request by NoaShapira8**
fixing bugs

### CODE REVIEW

File: src_py/apiServer/stats.py [hunk 1]
```diff
@@ -225,18 +225,36 @@ def build_worker_label_df(original_df, batch_ids, batch_size):
...
-                    max_column_labels_index = max_column_labels_index.tolist()
+                    #check if there is a sample with more than one predicted label
+                    max_in_row = df_worker_labels.iloc[:, num_of_labels:].max(axis=1)
+                    is_max = df_worker_labels.iloc[:, num_of_labels:].eq(max_in_row, axis=0)  #Get a DataFrame of boolean values where True indicates the maximum value in that row
+                    max_counts = is_max.sum(axis=1)    #Get the number of maximum values in each row
+                    has_multiple_max = max_counts.gt(1).any()  #boolean value: checks if there is at least one row with multiple maximum values in the predict labels
+
+                    if has_multiple_max:
+                        LOG_INFO(f"Worker {worker_name} has at least one sample with multiple predicted labels")
+                        #max_column_predict_index = is_max.apply(lambda row: list(row[row].index), axis=1).tolist()  # Generate a list of lists' each sublist has the index of the maximum value in the row
```
Comment (leondavi) [inline src_py/apiServer/stats.py:236]: Why this line is commented out. Decide to remove or to add it with condition.

File: src_py/apiServer/stats.py [hunk 2]
```diff
@@ -225,18 +225,36 @@ def build_worker_label_df(original_df, batch_ids, batch_size):
...
+                        max_column_predict_index = is_max.apply(lambda row: list(row[row].index) if row.any() else [-1], axis=1).tolist()
+                        max_column_predict_index =[[int(predict_label) - num_of_labels for predict_label in prdict_indexes_sublist] for prdict_indexes_sublist in max_column_predict_index]  # fix the index to original labels index
+                        max_column_labels_index = df_worker_labels.iloc[:, :num_of_labels].idxmax(axis=1).tolist()  # Get the index of the maximum actual value in each row
+
+                    else:            # No sample with multiple predicted labels
```
Comment (leondavi) [inline src_py/apiServer/stats.py:241]: What does it mean No sample?

File: src_py/apiServer/stats.py [hunk 3]
```diff
@@ -225,18 +225,36 @@ def build_worker_label_df(original_df, batch_ids, batch_size):
...
+                        # Take 2 list from the df, one for the actual labels and one for the predict labels to build the confusion matrix
```
Comment (leondavi) [inline src_py/apiServer/stats.py:242]: Typo - lists

File: src_py/apiServer/stats.py [hunk 4]
```diff
@@ -225,18 +225,36 @@ def build_worker_label_df(original_df, batch_ids, batch_size):
...
+                        if has_multiple_max:
+                            class_predict_list = [1 if class_index in row_max_list else 0 for row_max_list in max_column_predict_index]
+                        else:
+                            class_predict_list = [1 if label_num == class_index else 0 for label_num in max_column_predict_index]   # 1 if the label is belong to the class, 0 otherwise
                         class_actual_list = [1 if label_num == class_index else 0 for label_num in max_column_labels_index]   # 1 if the label is belong to the class, 0 otherwise
-                        class_predict_list = [1 if label_num == class_index else 0 for label_num in max_column_predict_index]   # 1 if the label is belong to the class, 0 otherwise
-                        confusion_matrix = metrics.confusion_matrix(class_actual_list, class_predict_list)  
+                        labels = [0, 1]
+                        confusion_matrix = metrics.confusion_matrix(class_actual_list, class_predict_list, labels=labels)  
                         #confusion_matrix_np = confusion_matrix.to_numpy()
```
Comment (leondavi) [inline src_py/apiServer/stats.py:258]: Decide - Remove if not needed

**Review by leondavi — CHANGES_REQUESTED** [review]

File: src_py/apiServer/stats.py (hunk 2 above)
Comment (NoaShapira8) [reply src_py/apiServer/stats.py:241]: that means that in each sample there is only 1 predicted label

File: src_py/apiServer/stats.py (hunk 1 above)
Comment (NoaShapira8) [reply src_py/apiServer/stats.py:236]: I removed it.

File: src_py/apiServer/stats.py (hunk 3 above)
Comment (NoaShapira8) [reply src_py/apiServer/stats.py:242]: ?

File: src_py/apiServer/stats.py (hunk 2 above)
Comment (leondavi) [reply src_py/apiServer/stats.py:240]: I still don't understand why no sample.

**Review by leondavi — APPROVED** [review]

**Pull request merged**